streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -169,8 +170,10 @@ public synchronized void start() {
             state = RUNNING;
 
             log.info("Started Kafka Stream process");
-        } else {
+        } else if (state == RUNNING) {
             throw new IllegalStateException("This process was already started.");
+        } else {
+            throw new IllegalStateException("Cannot restart after closing.");
```

Comment:
The existing error message here is only appropriate if `start()` had previously been called, but it's also possible for this to fail is `close()` is called before `start()`. There are also now tests for these two cases.